src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -145,6 +150,50 @@ private async Task OnResourcesPrepared(OnResourcesPreparedContext _)
         await PublishResourcesWithInitialStateAsync().ConfigureAwait(false);
     }
 
+    private Task ProcessResourcesWithoutLifetime(AfterEndpointsAllocatedEvent @event, CancellationToken cancellationToken)
```

Comment:
Should these resources raise the `ResourceReadyEvent`. There is a meta question here - its not that these resources don't have a lifetime - its more that their lifetime only has internal dependencies.

Reply:
These resources don't have a lifetime. Are you saying we should publish the event from here or something else? ResourceReadyEvent will never fire because these are never in a running state.

Reply:
In the playground sample in the parameter, the connection string references the DB resource which inherits its lifecycle from its parent. I'm not seeing this as much different other than its not strictly a "child" resource. I don't think that this is a decision we need to block on though, just something we need to consider that we may want to revisit.

Reply:
I stuffed up when debugging this to validate what I was saying. It DOES fire ResourceReadyEvent for SqlDatabases (as an example of a child resource).

Reply:
That's more of a dependency that inheriting the lifecycle (like WaitFor). It does make me wonder if enabling that does make it more interesting to support WaitFor.

Reply:
Updated WaitFor to handle this situation.